### ISSUE TITLE
Show warning when breaking design rules

### ISSUE THREAD

**Issue by Zefau** [ioBroker/ioBroker.js-controller#2040]
https://github.com/ioBroker/ioBroker.admin/issues/1772#issuecomment-1303038900
> From a developer perspective I would just appreciate certain rules and their enforcement. This topic is one of them. Another topic would be white spaces in ioBroker States / Objects.

> Jarvis rejects whitespaces but certain adapters create such states / objects.

> All adapters breaking such rules should at least be "punished" with a warning or error in the log. Wherever it comes from.

**Comment by Zefau** [comment]
> State will be displayed as folder, when it holds states itself, see https://github.com/Zefau/ioBroker.jarvis/issues/2010.

**Comment by Apollon77** [comment]
Lets check this hear:

> Jarvis rejects whitespaces but certain adapters create such states / objects.

Do you have an example? Normally there is a logic in js-controller which replaces all not officially allowed characters by "_" ...


For the general thing - the big question is how to "know" the full object structure to be able to write such infos for "object structure" topics in a performant way

**Comment by foxriver76** [comment]
@Apollon77 Yes, not official chars are replaced but white spaces are still allowed afaik
So this would be a debate about deprecating whitespaces which will probably break some legacy adapters. 

With states under states I am completely with it, it is a bad practice. With our naive approach it is a bit expensive to check. On object creation get one level up and check if it is a state, and then the problem arises what if one level up is no object and two levels up is state etc, or also disallow for one level up is nothing as it is also a bad practice? also later on Someone could change a folder to a state an then we would need to check that it has no childs, which will be even more expensive.

**Comment by Jey-Cee** [comment]
Not a solution for this topic, but a step to prevent the need of such a check: 
As a lot of adapter do not define the objects explicit while creating them dynamic, it is not possible to automate the check by adapter checker. 
A workaround could be that the developer has to add an export of the object structure to the repository. 
This makes it possible to review them automated.

**Comment by mcm1957** [comment]
> Not a solution for this topic, but a step to prevent the need of such a check: As a lot of adapter do not define the objects explicit while creating them dynamic, it is not possible to automate the check by adapter checker. A workaround could be that the developer has to add an export of the object structure to the repository. This makes it possible to review them automated.

I think this would be impossible whenever the states are created dynamically based on the actual device. I.e. an adapter checking an computer could  create an state for every harddisc. Or another example: the snmp adapetr cretes states based on user config (one for every configured oid). These states are not known at time of commit to repository. So ist impossible to add any info about them to the repository.
